src/include/duckdb/storage/checkpoint/string_checkpoint_state.hpp

```diff
@@ -65,8 +65,8 @@ struct UncompressedStringSegmentState : public CompressedSegmentState {
 	}
 
 private:
-	mutex block_lock;
-	unordered_map<block_id_t, shared_ptr<BlockHandle>> handles;
+	mutable mutex block_lock;
+	mutable unordered_map<block_id_t, shared_ptr<BlockHandle>> handles;
```

Comment:
Not sure why we're making the function `const` if we're making both fields `mutable`. Is this necessary?

Is it because `ColumnSegment` is now `const` in a bunch of places, and it has this field?
```c++
unique_ptr<CompressedSegmentState> segment_state;
```

I think that this should still work, since the `const` to `ColumnSegment` applied to `unique_ptr`, and not to `CompressedSegmentState`, so you would still be able to deref the pointer and get a non-const segment state.

Reply:
Hmm I'm not a huge fan of that, that is still changing internal state that is owned by the ColumnSegment, now it's just explicit that it's being changed in a method that is viewed as const

I don't really know what the implications of populating the `handles` here are.
I'm pretty sure the `block_lock` has to be `mutable` and that change is fine, we still want to ensure safe access even in const methods.

It's *possible* to remove the mutable from `handles`, if we change the `GetHandle() const` version to not lazily load the handle, returning nullptr if it doesn't exist, but I don't feel confident making that change